src/main/java/frc/robot/subsystems/TrapElvSubsystem.java

```diff
@@ -0,0 +1,420 @@
+    baseMotor2.getPIDController().setIZone(TrapElvConstants.BASE_PID[3]);
+    baseMotor2.getPIDController().setFF(TrapElvConstants.BASE_PID[4]);
+
+    scoringMotor = new CANSparkMaxSim(TrapElvConstants.SCORING_MOTOR_ID, MotorType.kBrushless);
+    scoringMotor.restoreFactoryDefaults();
+    scoringMotor.getPIDController().setP(TrapElvConstants.SCORING_PID[0]);
+    scoringMotor.getPIDController().setI(TrapElvConstants.SCORING_PID[1]);
+    scoringMotor.getPIDController().setD(TrapElvConstants.SCORING_PID[2]);
+    scoringMotor.getPIDController().setIZone(TrapElvConstants.SCORING_PID[3]);
+    scoringMotor.getPIDController().setFF(TrapElvConstants.SCORING_PID[4]);
+    TrapElvTab.add("Scoring Elv PID", scoringMotor.getPIDController());
+
+    // SmartDashboard
+    sourceLog = new DebugEntry<Boolean>(baseLimit.get(), "Source Beam Break", this);
+    groundLog = new DebugEntry<Boolean>(baseLimit.get(), "Ground Beam Break", this);
+    baseLog = new DebugEntry<Boolean>(baseLimit.get(), "Base Limit Switch", this);
+    scoringLog = new DebugEntry<Boolean>(baseLimit.get(), "Scoring Limit Switch", this);
+
+    // Simulation
+    if (Robot.isSimulation()) {
+      // 2D Mechanism
+      elvMechanism = new Mechanism2d(2, 2);
+      root = elvMechanism.getRoot("Root", 1, 0);
+      baseMech =
+          root.append(new MechanismLigament2d("Base Elv", 0, 90, 20, new Color8Bit(Color.kBlue)));
+      scoringMech =
+          baseMech.append(
+              new MechanismLigament2d("Scoring Elv", 0, 0, 15, new Color8Bit(Color.kAqua)));
+      wristMech =
+          scoringMech.append(
+              new MechanismLigament2d(
+                  "Wrist Mech", TrapElvConstants.WRIST_LENGTH, 0, 10, new Color8Bit(Color.kRed)));
+
+      m_baseElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(2),
+              TrapElvConstants.ELV_GEAR_RATIO,
+              TrapElvConstants.ELV_LIFT_MASS,
+              TrapElvConstants.DRUM_RADIUS,
+              TrapElvConstants.ELV_MIN_HEIGHT,
+              TrapElvConstants.ELV_MAX_HEIGHT,
+              true,
+              0);
+
+      m_scoringElevatorSim =
+          new ElevatorSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.ELV_GEAR_RATIO,
+              TrapElvConstants.ELV_LIFT_MASS,
+              TrapElvConstants.DRUM_RADIUS,
+              TrapElvConstants.ELV_MIN_HEIGHT,
+              TrapElvConstants.ELV_MAX_HEIGHT,
+              true,
+              0);
+
+      m_wristMotorSim =
+          new SingleJointedArmSim(
+              DCMotor.getNEO(1),
+              TrapElvConstants.WRIST_GEAR_RATIO,
+              (1.0 / 3.0)
+                  * TrapElvConstants.ELV_LIFT_MASS
+                  * Math.pow(TrapElvConstants.WRIST_LENGTH, 2.0),
+              TrapElvConstants.WRIST_LENGTH,
+              TrapElvConstants.WRIST_MIN_ANGLE, // min rotation
+              TrapElvConstants.WRIST_MAX_ANGLE, // max rotation
+              false,
+              0);
```

Comment:
Gravity for this physics model points in the -90 degree direction. 

For the mechanism, however, down is 180 degrees. 

Should be able to make those agree by adding offsets in the right places.